crates/starknet-devnet-types/src/rpc/transactions/declare_transaction_v0v1.rs

```diff
@@ -16,19 +12,18 @@ pub struct DeclareTransactionV0V1 {
     pub nonce: Nonce,
     pub max_fee: Fee,
     pub version: TransactionVersion,
-    pub transaction_hash: TransactionHash,
     pub signature: TransactionSignature,
 }
 
 impl DeclareTransactionV0V1 {
-    pub fn get_transaction_hash(&self) -> &TransactionHash {
-        &self.transaction_hash
-    }
-}
-
-impl HashProducer for DeclareTransactionV0V1 {
-    type Error = Error;
-    fn generate_hash(&self) -> DevnetResult<Felt> {
-        Ok(self.transaction_hash)
+    pub fn new(broadcasted_txn: &BroadcastedDeclareTransactionV1, class_hash: ClassHash) -> Self {
+        Self {
+            class_hash,
+            sender_address: broadcasted_txn.sender_address,
+            nonce: broadcasted_txn.common.nonce,
+            max_fee: broadcasted_txn.common.max_fee,
+            version: broadcasted_txn.common.version,
+            signature: broadcasted_txn.common.signature.clone(),
+        }
```

Comment:
This shouldn't present a slowdown caused by more often calling hash generation? It should be called only once?

Reply:
The transaction_hash is calculated when the object is converted to blockifier transaction only. So it will be only once yes